File: Implementation.java
import java.util.concurrent.*;
import java.io.*;
import java.util.*;

public class DataProcessingSystem {
    private static final BlockingQueue<Task> taskQueue = new LinkedBlockingQueue<>();
    private static final List<String> results = Collections.synchronizedList(new ArrayList<>());

    public static void main(String[] args) throws InterruptedException {
        ExecutorService executor = Executors.newFixedThreadPool(4);

        // Simulate adding tasks
        for (int i = 0; i < 10; i++) {
            taskQueue.put(new Task("Task " + i));
        }

        // Worker threads
        for (int i = 0; i < 4; i++) {
            executor.submit(new Worker(taskQueue, results));
        }

        executor.shutdown();
        executor.awaitTermination(1, TimeUnit.MINUTES);

        // Print results
        results.forEach(System.out::println);
    }
}

class Worker implements Runnable {
    private final BlockingQueue<Task> queue;
    private final List<String> results;

    public Worker(BlockingQueue<Task> queue, List<String> results) {
        this.queue = queue;
        this.results = results;
    }

    @Override
    public void run() {
        try {
            Task task = queue.take();
            // Simulate task processing
            Thread.sleep(1000);
            results.add("Processed: " + task.getName());
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }
}

class Task {
    private final String name;

    public Task(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: implementation.go
package main

import (
	"fmt"
	"sync"
	"time"
)

type Task struct {
	Name string
}

func main() {
	var wg sync.WaitGroup
	taskQueue := make(chan Task, 10)
	results := []string{}

	// Simulate adding tasks
	for i := 0; i < 10; i++ {
		taskQueue <- Task{Name: fmt.Sprintf("Task %d", i)}
	}

	// Worker goroutines
	for i := 0; i < 4; i++ {
		wg.Add(1)
		go worker(taskQueue, &results, &wg)
	}

	// Close channel after all tasks are added
	close(taskQueue)

	// Wait for all goroutines to finish
	wg.Wait()

	// Print results
	for _, result := range results {
		fmt.Println(result)
	}
}

func worker(taskQueue chan Task, results *[]string, wg *sync.WaitGroup) {
	defer wg.Done()
	for task := range taskQueue {
		// Simulate task processing
		time.Sleep(1 * time.Second)
		*results = append(*results, "Processed: "+task.Name)
	}
}
